<?xml version="1.0"?>
<launch>
  <include file="$(find caddy_bot)/launch/description.launch.xml"/>
  
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="50.0" />
    <param name="tf_prefix" value=""/>
  </node>

  <!-- <include file="$(find ntrip_ros)/launch/ntrip_ros.launch"/>
  <include file="$(find ublox_gps)/launch/ublox_device.launch"/>
  <node pkg="caddy_bot" type="gps_copy.py" name="gps_copy" clear_params="true"/> -->

  <node pkg="oroca_ros_tutorials" type="ros_tutorial_msg_publisher" name="ros_tutorial_msg_publisher" clear_params="true"/>
  <node pkg="caddy_bot" type="motor.py" name="motor" clear_params="true"/>

  <!-- <include file="$(find caddy_bot)/launch/test.launch"/> -->

  <include file="$(find e2box_imu)/launch/e2box_imu.launch"/>

  <include file="$(find realsense2_camera)/launch/rs_rgbd.launch">
    <arg name="align_depth" value="true"/>
    <arg name="depth_width" value="424"/>
    <arg name="depth_height" value="240"/>
    <arg name="depth_fps" value="30"/>
    <arg name="color_width" value="424"/>
    <arg name="color_height" value="240"/>
    <arg name="color_fps" value="30"/>
  </include>
  <include file="$(find depthimage_to_laserscan)/launch/launchfile_sample.launch"/>

  <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="base_footprint_caddy_to_odom" args="0 0 0 0 0 0 base_footprint_caddy odom"/> -->

  <rosparam command="load" file="$(find caddy_bot)/params/ekf_params.yaml" />
  <rosparam command="load" file="$(find caddy_bot)/params/navsat_params_sim.yaml" />

  <node pkg="robot_localization" type="ekf_localization_node" name="ekf_se_odom" clear_params="true"/>

  <node pkg="robot_localization" type="ekf_localization_node" name="ekf_se_map" clear_params="true">
    <remap from="odometry/filtered" to="odometry/filtered_map"/>
  </node>

  <node pkg="robot_localization" type="navsat_transform_node" name="navsat_transform" clear_params="true" output="screen" >
      <remap from="odometry/filtered" to="odometry/filtered_map"/>
      <!-- <remap from="gps/fix" to="ublox_gps/fix"/> -->
      <!-- <remap from="gps/fix" to="ublox_gps_copy"/> -->
      <remap from="gps/fix" to="vc_gps_caddy"/>
      <remap from="imu/data" to="imu_e2box"/>
  </node>

  <include file="$(find caddy_bot)/launch/move_base.launch"/>
  <!-- <node type="rviz" name="rviz" pkg="rviz" args="-d $(find caddy_bot)/rviz/DD.rviz" /> -->

</launch>
 